proto_helper/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+
+pub struct ProtoBuilder {
+    protoc_search_path: Option<PathBuf>,
+    files: Vec<PathBuf>,
+    import_paths: Vec<PathBuf>,
+    generate_grpc: bool,
+    transformations: Vec<Box<Fn(String) -> String>>,
+    grpc_transformations: Vec<Box<Fn(String) -> String>>,
+}
+
+impl ProtoBuilder {
+    // The path can be relative to the build script.
+    pub fn new() -> Self {
+        Self {
+            protoc_search_path: None,
+            files: vec![],
+            import_paths: vec![],
+            generate_grpc: false,
+            transformations: vec![],
+            grpc_transformations: vec![],
+        }
+    }
+
+    // The path to the directory where protoc is located. If this is not set, PATH is used.
+    pub fn protoc_search_path<P: AsRef<Path>>(&mut self, protoc_search_path: P) -> &mut Self {
+        self.protoc_search_path = Some(protoc_search_path.as_ref().to_owned());
+        self
+    }
+
+    // Register a file for compilation. The path must be relative to the build script.
+    pub fn add_file<P: AsRef<Path>>(&mut self, file_path: P) -> &mut Self {
+        self.files.push(file_path.as_ref().to_owned());
+        self
+    }
+
+    pub fn add_import_path<P: AsRef<Path>>(&mut self, import_path: P) -> &mut Self {
+        self.import_paths.push(import_path.as_ref().to_owned());
+        self
+    }
+
+    pub fn grpc(&mut self, generate_grpc: bool) -> &mut Self {
+        self.generate_grpc = generate_grpc;
+        self
+    }
+
+    pub fn add_transformation(&mut self, f: Box<Fn(String) -> String>) -> &mut Self {
```

Comment:
nit: I think it would be easier to use this and next methods if instead of taking a boxed function as an argument, we accept a generic function as an argument and box it on our own, i.e something like.
```
    pub fn add_transformation<F: Fn(String) -> String> (&mut self, f:  F) -> &mut Self {
        self.transformations.push(Box::new(f));
        self
    }
```

Reply:
F would need to be 'static, which seems a bit of an unpleasant restriction.

Reply:
ah, yes, I didn't take this into account.